consensus/ibft/ibft.go

```diff
@@ -1249,13 +1249,31 @@ func (i *Ibft) VerifyHeader(parent, header *types.Header) error {
 	}
 
 	// verify the committed seals
-	if err := verifyCommitedFields(snap, header); err != nil {
+	if err := verifyCommitedFields(snap, header, i.quorumSize(header.Number)); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func (i *Ibft) quorumSize(blockNumber uint64) QuorumImplementation {
+	rawUint64, ok := i.config.Config["quorumSizeBlockNum"]
+	if !ok {
+		return OptimalQuorumSize
+	}
+
+	cfgQuorumSizeBlockNum, ok := rawUint64.(float64)
+	if !ok {
+		panic("invalid block number format")
+	}
+
+	if blockNumber < uint64(cfgQuorumSizeBlockNum) {
+		return LegacyQuorumSize
+	} else {
+		return OptimalQuorumSize
+	}
```

Comment:
You can drop the `else` here, just have a 

`return OptimalQuorumSize`

Reply:
[Dropped](https://github.com/0xPolygon/polygon-edge/pull/549/commits/bdc844ddce828e630c441dfb5c29a32b0e417e64)